controllers/grafanafolder_controller.go

```diff
@@ -400,6 +405,30 @@ func (r *GrafanaFolderReconciler) onFolderCreated(ctx context.Context, grafana *
 	return nil
 }
 
+func (r *GrafanaFolderReconciler) retrieveParentFolderUID(ctx context.Context, cr *grafanav1beta1.GrafanaFolder) (string, error) {
+	if cr.Spec.ParentFolderRef != "" && cr.Spec.ParentFolderUID != "" {
```

Comment:
This check can be implemented on the CRD layer: https://github.com/grafana/grafana-operator/blob/master/api/v1beta1/grafanaalertrulegroup_types.go#L26

By doing it in the CRD layer, the user receives immediate feedback and we prevent invalid states

Reply:
Omg - I was trying to find how to do this for at least 3 hours by diving in kubebuilder deep issues. I was trying to do this at the field level and not at the object level (which was resulting to an error...).

I will go this one for sure.